reana_workflow_controller/consumer.py

```diff
@@ -118,11 +118,21 @@ def on_message(self, body, message):
                     )
             elif workflow and workflow.status not in ALIVE_STATUSES:
                 logging.warning(
-                    f"Event for not alive workflow {workflow.id_} with DB status {workflow.status} received:\n"
-                    f"{body}\nIgnoring..."
+                    f"Event for not alive workflow {workflow.id_} with DB status {workflow.status} received."
                 )
+                try:
+                    _delete_workflow_batch_pod(workflow)
+                    logging.info(
+                        f"Remove batch-pod for not alive {workflow.id_} workflow."
+                    )
+                except REANAWorkflowControllerError as exception:
+                    logging.error(
+                        f"Could not clean up not alive workflow {workflow.id_} batch pod for workflow."
+                        f" Error: {exception}"
+                    )
+                _delete_workflow_jobs(workflow)
```

Comment:
I have a problem in this line with deciding whether to delete workflow jobs or not. I am not sure how to test this scenario, and if this is the right place to remove jobs.

(1) Theoretically, it is possible that when the workflow is deleted but there are a lot of messages in the `jobs-status` queue, its deletion will be delayed. And during this delay, the workflow pod can start executing jobs. But, I have a hard time coming up with a test scenario (unit or manual).

(2) In addition, I am not sure if this is the right place to delete jobs. It would make more sense to delete them on a shutdown of `workflow pod`.

WDYT?

Reply:
If you are only addressing the `NotReady` situation for the workflow engine pod, then I was not seeing many situations in production where we needed to kill also workflow step jobs... So I think we should be mostly okay here already?

The situation of surviving workflow step pods happened in the past in other kind of troubles that were not related to `NotReady`. I recall one recent example where workflow step pods survived in the `Completed` status but there was no workflow batch pod anymore. That is, the workflow batch pod apparently terminated well, but some of the workflow step pods still remained in `Completed` status. Have you tried to look at this situation too?

FWIW I tend to agree with you about the danger of deleting things for all not `ALIVE_STATUSES` because that could include ones we don't want perhaps?

Reply:
Good point about not deleting `not ALIVE_STATUSES`. I should only delete workflows that have `deleted` status

Reply:
> Have you tried to look at this situation too?

Do we have an issue describing in detail this situation? I couldn't find it.

> I recall one recent example where workflow step pods survived in the Completed status but there was no workflow batch pod anymore.

Step jobs are removed in `job-controller` when they are finished. It probably means something failed in `job-controller`. There is a possibility to automate the step job deletion process in Kubernetes - https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/#ttl-after-finished-controller . But this is a discussion for a new issue.

Reply:
> If you are only addressing the NotReady situation for the workflow engine pod, then I was not seeing many situations in production where we needed to kill also workflow step jobs... So I think we should be mostly okay here already?

This is a tricky question. First of all, the issue with hanging step jobs is unlikely. So we are talking about an edge case here. 

In the situation above, If you wait long enough the only thing that would be left hanging is a workflow engine pod. In reality, before that, all step jobs would run, finish, and get deleted. So it wastes the resources of the cluster.